packages/SwingSet/docs/vat-environment.md

```diff
@@ -0,0 +1,113 @@
+
+As SES is on the JavaScript standards track, it includes other proposed features. If those features become standards, future JS environments will include them as global objects. So the current SES shim also makes those global objects available.
+
+## Additions
+
+`console` is available to vat code as a deliberate addition. It is not part of JavaScript, but it's so important for debugging that leaving it out would cause too much confusion. Note that the exact behavior of `console.log` is up to the host program: it is not guaranteed to be displayed to the operator. Applications that want to let vats write to `process.stdout` must somehow provide that authority directly. From a vat's point of view, `console` is frozen and powerless.
+
+`harden` is available as a global. It freezes the API surface (enumerable data properties) of an object. For more details, see [the `harden` package](https://github.com/Agoric/SES-shim/blob/SES-v0.8.0/packages/harden/README.md).
+
+`HandledPromise` is also a global. The `E` wrapper (`E(target).methodname(args)`) can be imported as `import { E } from '@agoric/eventual-send`. These are defined by the TC39 [Eventual-Send Proposal](https://github.com/tc39/proposal-eventual-send). In addition, "tildot" syntax (`target~.methodname(args)`) can be used by vat code.
+
+`Compartment`, [part of SES](https://github.com/Agoric/SES-shim/tree/SES-v0.8.0/packages/ses#compartment), is a global. Vat code runs inside a Compartment, but vat code can create sub-compartments to host other code (with different globals or transforms). Note that these child Compartments get `harden` and `Compartment`, but they won't include other endowments (like `console` or `HandledPromise`) unless you explicitly provide them. Child compartments will have the metering and tildot transformations applied (these are inescapable). Child compartments will *not* be frozen by default, so if you want to prohibit ambient communication channels through the global scope, you must call `harden(c.globalThis)` before loading any untrusted code into the new compartment.
+
+## Removals
+
+Nearly all existing JS code was written to run under Node.js or inside a browser, so it's easy to conflate the features of these environments with those of the core language itself. As a result, many JavaScript developers may be surprised to learn that e.g. `Buffer` and `require` are not actually a part of the language, but are instead additions made by Node.js.
+
+Most of the Node.js-specific [global objects](https://nodejs.org/dist/latest-v14.x/docs/api/globals.html) are not available within a vat:
```

Comment:
"within a vat. These include" (or "are" if it's all of them).  It took me a moment to connect this to the list, so make it clearer what the list is about.